### PR TITLE
Add buttons support + Use Categories

### PR DESCRIPTION
**Pull request by Crenshaw1312**
I saw you had plans to do this, but I got impatient and did it myself in like 20 minutes, I plan on improving it a bit more and matching your style so it's more seamless but it works fine atm/

I also made it use categories for images and the buttons so it's not stupidly long and a pain to scroll, to focus on this, the image category auto-closes if the buttons one is open and vice versa

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -37,6 +37,14 @@ module.exports = class customRPC extends Plugin {
 				small_text: this.settings.get('small_text', undefined) || undefined,
 			},
 		};
+
+		let buttons = [];
+		if (this.settings.get('button1', {label: "", url: ""}).label != "" && this.settings.get('button1', {label: "", url: ""}).url != "") buttons.push(this.settings.get('button1'));
+		if (this.settings.get('button2', {label: "", url: ""}).label != "" && this.settings.get('button2', {label: "", url: ""}).url != "") buttons.push(this.settings.get('button2'));
+		if (buttons[0]) rp.buttons = buttons;
+		console.log(buttons);
+		console.log(rp);
```
Comment (Vendicated) [inline index.js]: should probably remove these before merging

File: Settings.jsx [hunk 2]
```diff
@@ -1,8 +1,14 @@
 const { React, getModule, getModuleByDisplayName } = require('powercord/webpack');
-const { TextInput, SwitchItem } = require('powercord/components/settings');
+const { TextInput, SwitchItem, Category } = require('powercord/components/settings');
 const path = require('path');
+const { get } = require('http');
```
Comment (E-boi) [inline Settings.jsx]: This needs to be removed idk why that's there

File: Settings.jsx [hunk 3]
```diff
@@ -48,46 +54,118 @@ module.exports = class RPCSettings extends React.PureComponent {
...
+						onChange={val => {
+							let button2 = getSetting('button2', this.state.button2Def);
+							button2.label = val;
+							updateSetting('button2', button2);
+							if (button2.url != "") {
+								powercord.pluginManager.get(__dirname.split(path.sep).pop()).reloadRPC();
+							}
+						}}
+					>
+						Button 1 Text
```
Comment (E-boi) [inline Settings.jsx]: I think this is suppose to say "Button 2 Text"

File: Settings.jsx [hunk 4]
```diff
@@ -48,46 +54,118 @@ module.exports = class RPCSettings extends React.PureComponent {
...
+						onChange={val => {
+							let button2 = getSetting('button2', this.state.button2Def);
+							button2.url = val;
+							updateSetting('button2', button2);
+							if (button2.label != "") {
+								powercord.pluginManager.get(__dirname.split(path.sep).pop()).reloadRPC();
+							}
+						}}
+					>
+						Button 1 Url
```
Comment (E-boi) [inline Settings.jsx]: Same for this "Button 2 Url"

File: index.js [hunk 5]
```diff
@@ -37,6 +37,14 @@ module.exports = class customRPC extends Plugin {
 				small_text: this.settings.get('small_text', undefined) || undefined,
 			},
 		};
+
+		let buttons = [];
+		if (this.settings.get('button1', {label: "", url: ""}).label != "" && this.settings.get('button1', {label: "", url: ""}).url != "") buttons.push(this.settings.get('button1'));
```
Comment (E-boi) [inline index.js]: Why are these empty but in settings by default button 1 text is "powercord.dev" and the url is "https://powercord.dev"

**Review by E-boi — CHANGES_REQUESTED** [review]

**Pull request merged**